src/main/java/me/drepic/proton/redis/RedisChannel.java

```diff
@@ -16,16 +18,15 @@ public RedisChannel(MessageContext context, String recipient) {
 
     @Override
     public String toString(){
-
         return this.context.toContextString() + "." + this.recipient;
     }
 
     public static RedisChannel fromString(String s){
         String[] l = s.split("\\.");
-        if(l.length != 3){
+        if(l.length < 2 || l.length > 3){
```

Comment:
what was wrong with `!= 3`

Reply:
Ah, I see, you are trying to capture `"namespace.subject"`. Well your toString() will always add a remaining dot to give: `"namespace.subject."` if recipient is null, so I am not sure this is necessary

Reply:
String#split() does not count trailing empty strings.